### PR TITLE
Add code for Openfoam wrapper creation using GUI output

### PR DESCRIPTION
**Pull request by martinRenou**

### CODE REVIEW

File: simphony_ui/openfoam_wrapper_creation.py [hunk 1]
```diff
@@ -0,0 +1,80 @@
+import numpy as np
+from simphony.engine import openfoam_file_io, openfoam_internal
+from simphony.core.cuba import CUBA
+
+
+def create_openfoam_wrapper(openfoam_settings):
```
Comment (stefanoborini) [inline simphony_ui/openfoam_wrapper_creation.py:6]: Ok, time we move into the documentation domain.

We have a format we use for docs. I think it's the "google python". 

It would be like this
```
""" Creates the openfoam wrapper setup from the settings as provided by the model object

Parameters
----------
openfoam_settings : OpenfoamModel
    The traited model describing the openfoam parameters

Returns
-------
openfoam_wrapper : Wrapper
     an instance of the appropriate Wrapper object, properly configured.
"""
```
Pycharm is smart enough to detect this docstring, and will tell you in other code if you are passing the proper types (it will parse the types in the doc)

File: simphony_ui/openfoam_wrapper_creation.py [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+    # Set boundary conditions
+    openfoam_wrapper.BC[CUBA.VELOCITY] = {}
+    openfoam_wrapper.BC[CUBA.PRESSURE] = {}
+
+    inlet_BC = openfoam_settings.boundary_conditions.inlet_BC
+    outlet_BC = openfoam_settings.boundary_conditions.outlet_BC
+    walls_BC = openfoam_settings.boundary_conditions.walls_BC
+    front_and_back_BC = openfoam_settings.boundary_conditions.front_and_back_BC
+
+    openfoam_wrapper.BC[CUBA.VELOCITY]['inlet'] = \
```
Comment (stefanoborini) [inline simphony_ui/openfoam_wrapper_creation.py]: to reduce repetition, create a local var and then use that to refer to ['inlet'] etc. Then, assign it to `openfoam_wrapper.BC[CUBA.VELOCITY]`. Same for PRESSURE

File: simphony_ui/openfoam_wrapper_creation.py [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+        get_boundary_condition_description(
+            outlet_BC.velocity_boundary_condition)
+    openfoam_wrapper.BC[CUBA.VELOCITY]['walls'] = \
+        get_boundary_condition_description(
+            walls_BC.velocity_boundary_condition)
+    openfoam_wrapper.BC[CUBA.VELOCITY]['frontAndBack'] = \
+        get_boundary_condition_description(
+            front_and_back_BC.velocity_boundary_condition)
+
+    openfoam_wrapper.BC[CUBA.PRESSURE]['inlet'] = \
```
Comment (stefanoborini) [inline simphony_ui/openfoam_wrapper_creation.py]: same here

**Review by stefanoborini — CHANGES_REQUESTED** [review]

**Comment by martinRenou** [comment]
We now run some tests using the OpenfoamModel, resulting in an error in Travis "cannot connect to X server" for the commit 0baca49.
I'll try to fix it.

File: simphony_ui/openfoam_wrapper_creation.py [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+    pressure_BC['walls'] = get_boundary_condition_description(
+            walls_BC.pressure_boundary_condition)
+    pressure_BC['frontAndBack'] = get_boundary_condition_description(
+            front_and_back_BC.pressure_boundary_condition)
+    openfoam_wrapper.BC[CUBA.PRESSURE] = pressure_BC
+
+    return openfoam_wrapper
+
+
+def get_boundary_condition_description(bc):
```
Comment (stefanoborini) [inline simphony_ui/openfoam_wrapper_creation.py:78]: Needs docs here as well.

File: simphony_ui/openfoam_wrapper_creation.py [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+        return bc.type
+    elif bc.type == 'fixedGradient':
+        # The shape of the fixed gradient is (1, 3)
+        return bc.type, bc.fixed_gradient[0].tolist()
+    elif bc.type == 'fixedValue':
+        if type(bc.fixed_value) == np.ndarray:
+            return bc.type, bc.fixed_value[0].tolist()
+        else:
+            return bc.type, bc.fixed_value
+
```
Comment (stefanoborini) [inline simphony_ui/openfoam_wrapper_creation.py]: I wonder if it would make sense to raise in this case.

File: simphony_ui/tests/test_openfoam_wrapper_creation.py [hunk 6]
```diff
@@ -0,0 +1,88 @@
...
+        self.assertDictEqual(
+            create_openfoam_wrapper(self.openfoam_model).BC[
+                CUBA.PRESSURE],
+            {
+                'inlet': ('fixedValue', 0.008),
+                'outlet': ('fixedValue', 0.0),
+                'walls': 'zeroGradient',
+                'frontAndBack': 'empty'
+            }
+        )
```
Comment (stefanoborini) [inline simphony_ui/tests/test_openfoam_wrapper_creation.py:101]: Add a test for get_boundary_condition_description, testing when it works, and when it should throw an exception (e.g. normal case vs corner case)

**Review by stefanoborini — CHANGES_REQUESTED** [review]

File: simphony_ui/openfoam_wrapper_creation.py (hunk 4 above)
Comment (stefanoborini) [reply simphony_ui/openfoam_wrapper_creation.py:78]:
```
Raises
--------
ValueError
```

**Review by stefanoborini — CHANGES_REQUESTED** [review]

**Comment by stefanoborini** [comment]
Good stuff fixing the X server issue.

**Review by stefanoborini — APPROVED** [review]
